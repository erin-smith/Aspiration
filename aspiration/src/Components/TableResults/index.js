import React from "react";
import { Container, Row, Col } from "../Grid";


function TableResults(props) {

  return (
    <Container>
      <Row className="flex-wrap-reverse">
        <Col size="md-8">
          <h2 className="font-italic"> Topic:  {props.name}</h2>
        </Col>
        <Col size="md-4">
        </Col>
      </Row>
      <Row>
        <Col size="md-6">
          <p className="font-italic small">Starred by {props.stargazerCount}</p>
        </Col>
      </Row>

      <Row>
        <Col size="md-6">
          <h5>Related Topics (max 10)</h5>
        </Col>
      </Row>
      <Row>
        <Col size="12 sm-8 md-10">
          <ul>
            {props.relatedTopics.map( item => {
              return <li ><button id={item.name} onClick={props.itemClicked}>{item.name} starred by {item.stargazerCount} users</button></li>;
            })}
          </ul>
        </Col>
      </Row>
    </Container>
  );
}

export default TableResults;