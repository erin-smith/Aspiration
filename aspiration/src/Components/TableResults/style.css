.btn-container {
  float: right;
}
li {
  text-decoration: underline;
}

@media (max-width: 767px) {
  .btn-container {
    margin-bottom: 15px;
  }
}
