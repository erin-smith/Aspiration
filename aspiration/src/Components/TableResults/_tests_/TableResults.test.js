import * as React from "react";
import TableResults from "../index";
import renderer from "react-test-renderer";

describe("TableResults", () => {
  test("Should match the snapshot", () => {

    const testData = {name:"", stargazerCount:0, relatedTopics:[]};

    const tree = renderer.create(
        <TableResults 
            name={testData.name} 
            stargazerCount={testData.stargazerCount}
            relatedTopics={testData.relatedTopics}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
