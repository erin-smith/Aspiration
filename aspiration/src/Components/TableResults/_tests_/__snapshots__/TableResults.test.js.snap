// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableResults Should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-8"
    >
      <h2
        className="font-italic"
      >
         Topic:  
        
      </h2>
    </div>
    <div
      className="col-md-4"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <p
        className="font-italic small"
      >
        Starred by 
        0
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <h5>
        Related Topics (max 10)
      </h5>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12 col-sm-8 col-md-10"
    >
      <ul />
    </div>
  </div>
</div>
`;
